### PR TITLE
Using lint- and compileWithContext is perfectly fine

### PR DESCRIPTION
**Pull request by schalkms**
Hence, I removed the comment.

### CODE REVIEW

File: detekt-test/src/main/kotlin/io/gitlab/arturbosch/detekt/test/RuleExtensions.kt [hunk 1]
```diff
@@ -38,11 +38,6 @@ fun BaseRule.lint(path: Path): List<Finding> {
     return findingsAfterVisit(ktFile)
 }
 
-/**
- * Compare with [compileAndLintWithContext], this does not use [KotlinScriptEngine]
- * to compile. Please use this sparingly: A typical usecase is when KotlinScript
- * failed to compile even when the Kotlin code in [content] is legal.
- */
 fun BaseRule.lintWithContext(
```
Comment (schalkms) [inline detekt-test/src/main/kotlin/io/gitlab/arturbosch/detekt/test/RuleExtensions.kt:41]: This function is now only used once by `compileAndLintWithContext()`.

**Comment by BraisGabin** [comment]
Don't we want to use compile when possible?

**Comment by schalkms** [comment]
> Don't we want to use compile when possible?

That's not possible. If top level are declared with visibility modifiers the Kotlin script engine cannot compile it.

**Review by BraisGabin — APPROVED** [review]
Oh! I though this was a comment in the lint extensions. 🤐

**Pull request merged**